### PR TITLE
Implement ObjectManager::unloadAll

### PR DESCRIPTION
**Pull request by duncanspumpkin**

### CODE REVIEW

File: src/OpenLoco/src/Objects/ObjectManager.cpp [hunk 1]
```diff
@@ -832,7 +832,19 @@ namespace OpenLoco::ObjectManager
     // 0x00472031
     void unloadAll()
     {
-        call(0x00472031);
+        for (auto type = ObjectType::interfaceSkin; enumValue(type) <= enumValue(ObjectType::scenarioText); type = static_cast<ObjectType>(enumValue(type) + 1))
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Objects/ObjectManager.cpp:835]: I get what you're doing... But wouldn't it be more readable to just iterate an integer to `kMaxObjectTypes` instead? Then cast that into an ObjectType when calling `getMaxObjects`.

File: src/OpenLoco/src/Objects/ObjectManager.cpp [hunk 2]
```diff
@@ -832,7 +832,19 @@ namespace OpenLoco::ObjectManager
...
+        {
+            for (LoadedObjectId id = 0; id < getMaxObjects(type); id++)
+            {
+                LoadedObjectHandle handle{ type, id };
+                auto* obj = getAny(handle);
```
Comment (LeftofZen) [inline src/OpenLoco/src/Objects/ObjectManager.cpp:840]: dont' really need the temporary `obj` variable

File: src/OpenLoco/src/Objects/ObjectManager.cpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/OpenLoco/src/Objects/ObjectManager.cpp]: okay i've reworked things

File: src/OpenLoco/src/Objects/ObjectManager.h [hunk 3]
```diff
@@ -194,4 +194,22 @@ namespace OpenLoco::ObjectManager
     void sub_4748FA();
 
     void registerHooks();
+
+    // Calls function with the handle (LoadedObjectHandle) of each loaded object
+    template<typename Function>
+    void forAllLoadedObjects(Function&& func)
```
Comment (ZehMatt) [inline src/OpenLoco/src/Objects/ObjectManager.h:200]: Minor nitpick, I would call it `forEachLoadedObject` which is more in line with a lot of algorithms.

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**